src/vs/vscode.d.ts

```diff
@@ -5400,6 +5400,26 @@ declare module 'vscode' {
 	}
 
 	/**
+	 * An event describing the change in Configuration
+	 */
+	export interface ConfigurationChangeEvent {
+
+		/**
+		 * Affected configuration keys
+		 */
+		readonly affectedKeys: string[];
```

Comment:
I agree with @jrieken, although I would tweak it a bit to be `changes: Map<string, { previous: any, current: any }>` if possible.